packages/hooks/src/useIsP2PEnabled.ts

```diff
@@ -1,22 +1,43 @@
+import React from 'react';
+import Cookies from 'js-cookie';
+import { deriv_urls } from '@deriv/shared';
 import { useStore } from '@deriv/stores';
-import useP2PConfig from './useP2PConfig';
+import useP2PSettings from './useP2PSettings';
 
 const useIsP2PEnabled = () => {
+    const {
+        p2p_settings,
+        rest: { isLoading, isSubscribed },
+    } = useP2PSettings();
     const { client, traders_hub } = useStore();
     // Todo: to replace it with useAuthorize hook
-    const { currency, is_virtual } = client;
+    const { currency, email, is_virtual, loginid, setIsP2PEnabled } = client;
     const { is_low_risk_cr_eu_real } = traders_hub;
-    const { data, ...rest } = useP2PConfig();
 
-    const is_p2p_supported_currency = Boolean(data?.supported_currencies.includes(currency.toLocaleLowerCase()));
+    const is_p2p_supported_currency = Boolean(
+        p2p_settings?.supported_currencies?.includes(currency.toLocaleLowerCase())
+    );
     const is_p2p_enabled = is_p2p_supported_currency && !is_virtual && !is_low_risk_cr_eu_real;
 
     // Todo: should replace with the next line instead once BE is fixed.
     // const is_p2p_enabled = data?.disabled === 0;
 
+    React.useEffect(() => {
+        setIsP2PEnabled(is_p2p_enabled);
+
+        if (email && loginid) {
+            const domain = /deriv\.(com|me)/.test(window.location.hostname)
+                ? deriv_urls.DERIV_HOST_NAME
+                : window.location.hostname;
+
+            Cookies.set('is_p2p_disabled', (!is_p2p_enabled).toString(), { domain, secure: true, sameSite: 'none' });
+        }
+    }, [email, is_p2p_enabled, loginid, setIsP2PEnabled]);
+
```

Comment:
previously, we set the cookie inside of socket-general, but to remove the dependency of client-store is_p2p_enabled value, moved it here instead when useIsP2PEnabled hook is called.

This is also where it sets the client-store value of is_p2p_enabled.